src/main/webapp/app/shared/range-slider/range-slider.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+
+const DEFAULT_STEP = 1;
+
+@Component({
+    selector: 'jhi-range-slider',
+    templateUrl: './range-slider.component.html',
+    styleUrls: ['./range-slider.component.scss'],
+    standalone: true,
+    imports: [FormsModule, ReactiveFormsModule],
+})
+export class RangeSliderComponent implements OnInit, OnDestroy {
+    @Input() generalMaxValue: number;
+    @Input() generalMinValue: number;
+    @Input() step: number = DEFAULT_STEP;
+
+    /** When extending the supported label symbols you might have to adjust the logic for */
+    @Input() labelSymbol?: '%';
+
+    @Input() selectedMinValue: number;
+    @Input() selectedMaxValue: number;
+    @Output() selectedMinValueChange: EventEmitter<number> = new EventEmitter<number>();
+    @Output() selectedMaxValueChange: EventEmitter<number> = new EventEmitter<number>();
+
+    rangeInputElements?: NodeList;
+    eventListeners: { element: HTMLInputElement; listener: (event: Event) => void }[] = [];
+
+    get sliderMinPercentage(): number {
+        return ((this.selectedMinValue - this.generalMinValue) / (this.generalMaxValue - this.generalMinValue)) * 100;
+    }
+
+    get sliderMaxPercentage(): number {
+        return 100 - ((this.selectedMaxValue - this.generalMinValue) / (this.generalMaxValue - this.generalMinValue)) * 100;
+    }
```

Comment:
It is now fixed by updating the respective value on the `input` hook - we do now only re-calculate the percentages when the respective input value changes and not on arbitrary change detection cycles.

With `ngOnChanges` the progress bar would have a "lagging" feeling as it will only be updated once the thumb is stopped being moved which is why I have chosen the `input` hook approach